crates/driver/src/tests/cases/protocol_fees.rs

```diff
@@ -50,23 +52,54 @@ async fn protocol_fee_test_case(test_case: TestCase) {
         .sell_amount(test_case.execution.solver.sell)
         .buy_amount(test_case.execution.solver.buy);
     let pool = ab_adjusted_pool(quote);
+    // Check if the order is expected to be partially filled by calculating a
+    // difference between solver's and order's target amounts
+    let partially_executed = match test_case.order.side {
+        order::Side::Sell => test_case
+            .order
+            .sell_amount
+            .saturating_sub(test_case.execution.solver.sell),
+        order::Side::Buy => test_case
+            .order
+            .buy_amount
+            .saturating_sub(test_case.execution.solver.buy),
+    };
+    // If there is a difference, the order is considered to be partially fillable
+    let partial = if partially_executed <= eth::U256::zero() {
+        Partial::No
+    } else {
+        Partial::Yes {
+            executed: partially_executed,
+        }
+    };
+    let solver_fee = test_case.execution.driver.sell / 100;
+    // Target amount to be executed by the solver in case of partially fillable
+    // order
+    let executed = match partial {
+        Partial::Yes { .. } => match test_case.order.side {
+            order::Side::Buy => Some(test_case.execution.solver.buy),
+            order::Side::Sell => Some(test_case.execution.solver.sell - solver_fee),
```

Comment:
Why do we subtract the solver fee here? Shouldn't the whole partial sell amount be executed?